src/litdata/streaming/reader.py

```diff
@@ -397,9 +401,19 @@ def read(self, index: ChunkedIndex) -> Any:
         chunk_filepath, begin, filesize_bytes = self.config[index]
 
         if isinstance(self._item_loader, PyTreeLoader):
-            item = self._item_loader.load_item_from_chunk(
-                index.index, index.chunk_index, chunk_filepath, begin, filesize_bytes, self._encryption
-            )
+            if self.no_chunk_download and self._config and self._config._remote_dir:
+                # we need to download relevant bytes only
+                # then deserialize them, and reconstruct the item with pytree's unflatten function
+                offset = (1 + (index.index - begin) if index.index >= begin else index.index + 1) * 4
+                pair = self.config.download_chunk_bytes_from_index(index.chunk_index, offset, 8)
+                begin, end = np.frombuffer(pair, np.uint32)
+                actual_item_length = end - begin
+                raw_bytes = self.config.download_chunk_bytes_from_index(index.chunk_index, begin, actual_item_length)
+                item = self._item_loader.load_item_from_bytes(raw_bytes, index.chunk_index)
```

Comment:
I think it would be nice to keep this entire part inside the item loader. wdyt ?